README.md

```diff
@@ -52,42 +52,291 @@ With most carrying a footer containing:
-* `{command: social, friends: <list of ALL friends>}`: Update the friends on the db
-* `{command: social, foes: <list of ALL foes>}`: Update the foe (muted players) on the db
+    {
+        "command": "ask_session"
+    }
 
-##### Avatar
-* `{command: avatar, action: upload_avatar, name: <avatar_name>, file: <file_content>, description: <desc>}`: Admin Command to upload an avatar
-* `{command: avatar, action: list_avatar}`: Send a list of available avatars
-* `{command: avatar, action: select, avatar: <avatar_url>}`: Select a valid avatar for the player
+#### Response
 
-##### Misc
+    {
+        "command": "session",
+        "session": long
+    }
 
-* [deprecated] `{command: ask_session}`: response with an welcome command and an valid session (can be delayed)
-* `{command: fa_state, state: <on|...>}`: notify the server if the game has launched or closed
-* `{command: quit_team}`: Leave a team
-* `{command: accept_team_proposal, leader: <leader_name>}`: Accept Team Invitation
-* `{command: hello, version: <...>, login: <...>, password: <...>, unique_id: <...>, (session: <...>)}`: Accept Team Invitation
+| Parameter | Value | Description              |
+|-----------|-------|--------------------------|
+| `session` | long  | The assigned session ID. |
 
-##  Stream
 
-The stream API is deprecated, but currently the following message types are supported:
 
-* `PING`: response with a `PONG`
-* `PONG`: internal state changed to ponged
-* `UPLOAD_MOD, login, session, zipmap, infos, size, fileDaatas`: Upload a mod
-* `UPLOAD_MAP, login, session, zipmap, infos, size, fileDatas`: Upload a map
+
+
+
+## Login
+
+#### Request
+
+Sends a login request.
+
+**TODO*** (Downlord:) I guess `local_ip` is not needed?
+**TODO*** (Downlord:) `username` instead of `login`? 
+**TODO*** (Downlord:) `password` plaintext in future? 
+
+    {
+        "login": string,
+        "password": string,
+        "session": long,
+        "unique_id": string,
+        "local_ip": string
+    }
+
+| Parameter   | Value  | Description                                                                                |
+|-------------|--------|--------------------------------------------------------------------------------------------|
+| `login`     | string | The username to log in.                                                                    |
+| `password`  | string | The SHA-256 hashed password.                                                               |
+| `session`   | long   | The ID of the current session.                                                             |
+| `unique_id` | string | The UUID generated by `uid.dll`                                                            |
+| `local_ip`  | string | The IP address of the local machine.<br/> *__Deprecated__: this may be removed in future*  |
+
+#### Response
+
+##### On success
+
+**TODO*** (Downlord:) `login` seems pretty useless; the client should know what we logged in with.
```

Comment:
Logins are case insensitive, it will work if the client attemps to login with a wrong casing.

The client needs to know which casing is correct for display purposes.